gradle-suppressible-error-prone/src/test/groovy/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePluginIntegrationTest.groovy

```diff
@@ -0,0 +1,531 @@
+        def sourceSet2 = new File(sourceSetRoot, '/build/somePlace')
+
+        writeJavaSourceFile(erroringCode, sourceSet1)
+        writeJavaSourceFile(erroringCode.replace('App', 'App2'), sourceSet2)
+
+        buildFile << """
+            sourceSets.main.java.srcDirs('${projectDir.relativePath(sourceSet1)}', '${projectDir.relativePath(sourceSet2)}')
+        """.stripIndent(true)
+
+        then:
+        runTasksSuccessfully('compileJava')
+    }
+
+    def 'can apply patches for a check if added to the patchChecks list'() {
+        // language=Gradle
+        buildFile << '''
+            suppressibleErrorProne {
+                patchChecks.add('ArrayToString')
+            }
+        '''.stripIndent(true)
+
+        // language=Java
+        writeJavaSourceFile '''
+            package app;
+            public final class App {
+                public static void main(String[] args) {
+                    System.out.println(new int[3].toString());
+                }
+            }
+        '''.stripIndent(true)
+
+        when:
+        runTasksSuccessfully('compileJava', '-PerrorProneApply')
+
+        then:
+        runTasksSuccessfully('compileJava')
+
+        appJava.text.contains('Arrays.toString(new int[3])')
+    }
+
+    def 'does not apply patches for a check if not added to the patchChecks list'() {
+        // language=Gradle
+        buildFile << '''
+            suppressibleErrorProne {
+                // To make sure set is not empty
+                patchChecks = ['SomeCheck']
+            }
+        '''.stripIndent(true)
+
+        // language=Java
+        writeJavaSourceFile '''
+            package app;
+            public final class App {
+                public static void main(String[] args) {
+                    System.out.println(new int[3].toString());
+                }
+            }
+        '''.stripIndent(true)
+
+        when:
+        runTasksSuccessfully('compileJava', '-PerrorProneApply')
+
+        then:
+        appJava.text.contains('new int[3].toString()')
+    }
+
+    def 'does not apply patches if there is nothing in patchChecks set'() {
+        // language=Gradle
+        buildFile << '''
+            suppressibleErrorProne {
+                patchChecks.empty()
+            }
+        '''.stripIndent(true)
+
+        // language=Java
+        writeJavaSourceFile '''
+            package app;
+            public final class App {
+                public static void main(String[] args) {
+                    System.out.println(new int[3].toString());
+                }
+            }
+        '''.stripIndent(true)
+
+        when:
+        // Doesn't actually do any patching as the set is empty. It just does a normal compile that fails.
+        def stderr = runTasksWithFailure('compileJava', '-PerrorProneApply').standardError
+
+        then:
+        stderr.contains('[ArrayToString]')
+        appJava.text.contains('new int[3].toString()')
```

Comment:
I feel like this is confusing behaviour from my above comment, it'll likely not matter as this isn't exactly for typical consumers